### PR TITLE
DSG-1666: create schema builder basic attributes

### PR DESCRIPTION
**Pull request by jgnuvalence**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

Ticket: https://dol-ewf.atlassian.net/browse/DSG-1666

The ability to drag and drop the basic schema attributes is added, these are:

1. String
2. Boolean
3. Integer
4. BigDecimal
5. LocalDate
6. LocalTime

## Screenshot of the affected changes

https://github.com/Nuvalence/dsgov-web/assets/97063738/a24a67fa-a44a-4909-9c8e-7f0a294d0c84

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/shared/feature/schema-nuv/src/lib/components/base/formio/formio-attribute-base.model.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+export enum AttributeTypes {
+  STRING = 'String',
+  BOOLEAN = 'Boolean',
+  LOCALDATE = 'LocalDate',
+  LOCALTIME = 'LocalTime',
+  DOCUMENT = 'Document',
+  LIST = 'List',
+  SCHEMA = 'Schema',
```
Comment (asultan001) [inline libs/shared/feature/schema-nuv/src/lib/components/base/formio/formio-attribute-base.model.ts:15]: I think schema and dynamicEntity are supposed to be the same but not sure which one is the correct one.

Comment (jgnuvalence) [reply libs/shared/feature/schema-nuv/src/lib/components/base/formio/formio-attribute-base.model.ts:15]: you are right, I just forgot to remove schema, I'll push this change along with the change in the attribute box width requested by you, I added a screenshot in the ticket, could you check it?

Comment (jgnuvalence) [reply libs/shared/feature/schema-nuv/src/lib/components/base/formio/formio-attribute-base.model.ts]: I saw your approval to change the attribute box, I just committed that and the removal of schema from this list

File: libs/shared/feature/schema-nuv/src/lib/components/base/formio/formio-attribute-base.model.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  LOCALDATE = 'LocalDate',
+  LOCALTIME = 'LocalTime',
+  DOCUMENT = 'Document',
+  LIST = 'List',
+  DYNAMICENTITY = 'DynamicEntity',
+  INTEGER = 'Integer',
+  BIGDECIMAL = 'BigDecimal',
+}
+
+export const typeToIconMap: Map<AttributeTypes, string> = new Map([
```
Comment (davidyoung77) [inline libs/shared/feature/schema-nuv/src/lib/components/base/formio/formio-attribute-base.model.ts:20]: Should we update the existing formio models to use this?

**Review by davidyoung77 — APPROVED** [review]
This all looks good thanks, nice work, we will likely rename the library to dynamic-schemas in the future but this looks good

**Pull request merged**